src/Builder/Encoder/OperatorEncoder.php

```diff
@@ -108,10 +106,19 @@ private function encodeAsSingle(OperatorInterface $value): stdClass
         throw new LogicException(sprintf('Class "%s" does not have a single property.', $value::class));
     }
 
+    /**
+     * Wrap the result in an object if the operator has a name.
+     * The operator name is NULL, and the result is returned as is
+     * when wrapObject is false in the YAML configuration of the operator.
+     */
     private function wrap(OperatorInterface $value, mixed $result): stdClass
```

Comment:
The current `mixed $result` type hint conflicts with the `stdClass` return type. I assume this method is already used elsewhere, so it's not feasible to simply change the parameter type.

Would it be easier to just inline the necessary logic in `encodeAsObject()` instead of modifying it use this existing method?

Reply:
Done, I also added a validation in `OperatorDefinition` to validate the `wrapObject` is set to false only on `object` encoding.